napalm_junos/junos.py

```diff
@@ -616,8 +616,7 @@ def get_lldp_neighbors_detail(self, interface=''):
             return {}
         interfaces = lldp_table.get().keys()
 
-        old_junos = napalm_base.helpers.convert(
-            int, self.device.facts.get('version', '0.0').split('.')[0], 0) < 13
+        old_junos = self.device.facts.get('personality') in ['SWITCH', 'SRX_HIGHEND', 'SRX_BRANCH']
 
         lldp_table.GET_RPC = 'get-lldp-interface-neighbors'
         if old_junos:
```

Comment:
As J series are not covered yet by junos-eznc personality, I would suggest the following (that will not affect the tests):

```python
if self.device.facts.get('personality') not in ('M', 'MX', 'T'):
```

And please add a comment with the references you mentioned earlier so it will make it obvious why we're treating MX series and SRX different.